Implement minStateScalar function to handle scalar operands in mask analysis

zhang-cx opened the pull request:
**Description:**
Currently, the mask analysis fails when either the left or right operand is a scalar. This PR introduces the minStateScalar function to address this issue, enabling the analysis to correctly process scalar operands.
With this patch, the analysis can now handle cases where one of the operands is a scalar or both of the operands are scalar, preventing the failure that previously occurred. As a result, tt.load can be successfully converted to tts.load.

**Changes:**
Added minStateScalar function to handle scalar operands in mask analysis
Updated mask analysis to utilize the new minStateScalar function
LIT test to test the new function

nhat-nguyen commented:
Thank you! The new lit test is failing because we use `triton-shared-opt` instead of `%triton-opt`. Same for `FileCheck` (no `%` prefix).

nhat-nguyen commented:
Also would you mind adding the triton program corresponding to the IR in the lit test as comments too if you have it? That will help with readability a bit. Otherwise patch looks good to me.

zhang-cx commented:
> Also would you mind adding the triton program corresponding to the IR in the lit test as comments too if you have it? That will help with readability a bit. Otherwise patch looks good to me.

@nhat-nguyen  Thanks for the advice! I added the original triton function and modified the command.

nhat-nguyen commented:
@zhang-cx Looks good! The CLA check is still failing though. I believe you have to do `@microsoft-github-policy-service agree company="Meta"`. Your previous reply wrapped the company with square brackets so maybe that didn't work.

zhang-cx commented:
@microsoft-github-policy-service agree company="Meta"
